UI/log-redaction.hpp

```diff
@@ -0,0 +1,16 @@
+#include <stdio.h>
+#include <regex>
+
+using namespace std;
+
+const regex login_url_regex =
+	regex("//[^\\s]+?:[^\\s]+?@([^\\s]+?)", regex_constants::icase);
+const regex macos_user_regex = regex("/Users/.+?/", regex_constants::icase);
+const regex linux_user_regex = regex("/home/.+?/", regex_constants::icase);
+const regex win_user_regex =
+	regex("[A-Za-z]:([\\\\])Users[\\\\].+?[\\\\]", regex_constants::icase);
+const regex rtmps_regex = regex("(rtmps?://[^\\s]+?/)[^\\s]+");
+const regex srt_regex = regex("(srt://.+?\\?).+?(?=$|\\s)");
+const regex connect_addr_regex = regex("Possible connect address: .+");
```

Comment:
Can this be `constexpr`? Also should they even be in the header? Should there even be a header?

Reply:
Constexpr isn't allowed for regex it appears. 

I thought there was some chance that these expressions might be useful elsewhere by themselves so I included a header. Could certainly be combined.

Reply:
> Can this be `constexpr`?

constexpr regex was proposed for C++23, but slipped to C++26, it seems.